File: Project 9/Java Version/LinkedListStack.java
import java.io.FileWriter;
import java.io.IOException;

public class LinkedListStack {
	private ListNode top;
	
	LinkedListStack(){
		this.top = null;
	}

	public ListNode getTop() {
		return top;
	}

	public void setTop(ListNode top) {
		this.top = top;
	}
	
	public void push(ListNode newNode) {
		newNode.setNext(top);
		top = newNode;
	}
	
	public ListNode pop() {
		if(top == null) {
			return null;
		}
		else {
			ListNode remove = top;
			top = top.getNext();
			remove.setNext(null);
			return remove;
		}
	} 
	
	public boolean isEmpty() {
		return (top == null);
	}
	
	public void printStack(ListNode top, FileWriter output) {
		String str = "Top --> ";
		ListNode node = top;
		try {
			for(node = top; node.getNext() != null; node = node.getNext()) {
				str += "(" + node.getData() + ", "+ node.getNext().getData() + ") --> ";
			}
			str += "(" + node.getData() + ", NULL) --> NULL\n\n";
			output.write(str);
		}
		catch(IOException e) {
			e.printStackTrace();
		}
		
	}
	
	
	
}

File: Project 9/C++ Version/LinkedListStack.cpp
#include "LinkedListStack.h"

LinkedListStack::LinkedListStack() {
	this->top = NULL;
}

LinkedListStack::~LinkedListStack(){
}

ListNode* LinkedListStack::getTop() {
	return top;
}

void LinkedListStack::setTop(ListNode* top) {
	this->top = top;
}

void LinkedListStack::push(ListNode* newNode) {
	newNode->setNext(top);
	top = newNode;
}

ListNode* LinkedListStack::pop() {
	if (top == NULL) {
		return NULL;
	}
	else {
		ListNode* remove = top;
		top = top->getNext();
		remove->setNext(NULL);
		return remove;
	}
}

bool LinkedListStack::isEmpty() {
	return (top == NULL);
}

void LinkedListStack::printStack(ListNode* top, ofstream& output) {
	string str = "Top --> ";
	ListNode* node = top;
	try {
		for (node = top; node->getNext() != NULL; node = node->getNext()) {
			str.append("(" + to_string(node->getData()));
			str.append(", " + to_string(node->getNext()->getData()));
			str.append(") --> ");
		}
		str.append("(" + to_string(node->getData()));
		str.append(", NULL) --> NULL\n\n");
		output << str;
	}
	catch (exception e) {
		cout << "There is an error: " << e.what() << endl;
	}
}